### PR TITLE
Remove obsolete useGDI code path and FontRenderingMode enum

### PR DESCRIPTION
**Pull request by fujii**
#### 21e5a249467d6356fa3cdacb6f49e12b542850fc
<pre>
Remove obsolete useGDI code path and FontRenderingMode enum
<a href="https://bugs.webkit.org/show_bug.cgi?id=206273">https://bugs.webkit.org/show_bug.cgi?id=206273</a>

Reviewed by Alex Christensen.

These code was used only by AppleWin port.

Define cWindowsFontScaleFactor to remove the magic number 32.

* Source/WebCore/Headers.cmake:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/page/DragController.cpp:
(WebCore::DragController::startDrag):
* Source/WebCore/page/Settings.yaml:
* Source/WebCore/page/SettingsBase.h:
* Source/WebCore/platform/DragImage.cpp:
(WebCore::createDragImageForLink):
* Source/WebCore/platform/DragImage.h:
* Source/WebCore/platform/cocoa/DragImageCocoa.mm:
(WebCore::createDragImageForLink):
* Source/WebCore/platform/graphics/FontCascade.h:
(WebCore::FontCascade::renderingMode const): Deleted.
* Source/WebCore/platform/graphics/FontCascadeCache.h:
(WebCore::FontDescriptionKey::makeFlagsKey):
* Source/WebCore/platform/graphics/FontDescription.cpp:
* Source/WebCore/platform/graphics/FontDescription.h:
(WebCore::FontDescription::fontSelectionRequest const):
(WebCore::FontDescription::setWeight):
(WebCore::FontDescription::operator== const):
(WebCore::FontDescription::encode const):
(WebCore::FontDescription::decode):
(WebCore::FontDescription::renderingMode const): Deleted.
(WebCore::FontDescription::setRenderingMode): Deleted.
* Source/WebCore/platform/graphics/FontPlatformData.h:
(WebCore::FontPlatformData::hfont const):
(WebCore::FontPlatformData::useGDI const): Deleted.
(WebCore::FontPlatformData::cgFont const): Deleted.
* Source/WebCore/platform/graphics/win/ComplexTextControllerUniscribe.cpp:
(WebCore::ComplexTextController::collectComplexTextRunsForCharacters):
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
(WebCore::FontCache::lastResortFallbackFont):
(WebCore::createGDIFont):
(WebCore::FontCache::createFontPlatformData):
* Source/WebCore/platform/graphics/win/FontCustomPlatformData.cpp: Removed.
* Source/WebCore/platform/graphics/win/FontCustomPlatformDataWin.cpp:
(WebCore::FontCustomPlatformData::fontPlatformData):
* Source/WebCore/platform/graphics/win/FontPlatformDataCairoWin.cpp:
(WebCore::FontPlatformData::platformDataInit):
(WebCore::FontPlatformData::platformIsEqual const):
* Source/WebCore/platform/graphics/win/FontPlatformDataWin.cpp:
(WebCore::FontPlatformData::FontPlatformData):
* Source/WebCore/platform/graphics/win/SimpleFontDataCairoWin.cpp:
(WebCore::Font::platformInit):
* Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp:
(WebCore::Font::platformCharWidthInit):
(WebCore::Font::platformCreateScaledFont const):
* Source/WebCore/platform/gtk/DragImageGtk.cpp:
(WebCore::createDragImageForLink):
* Source/WebCore/platform/ios/DragImageIOS.mm:
(WebCore::createDragImageForLink):
* Source/WebCore/platform/text/FontRenderingMode.h: Removed.
* Source/WebCore/platform/win/DragImageWin.cpp:
(WebCore::dragLabelFont):
(WebCore::createDragImageForLink):
* Source/WebCore/rendering/RenderEmbeddedObject.cpp:
(WebCore::RenderEmbeddedObject::getReplacementTextGeometry const):
* Source/WebCore/style/StyleResolveForDocument.cpp:
(WebCore::Style::resolveForDocument):
* Source/WebCore/style/StyleResolver.cpp:
(WebCore::Style::Resolver::defaultStyleForElement):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/Shared/win/WebCoreArgumentCodersWin.cpp:
(IPC::ArgumentCoder&lt;Font&gt;::decodePlatformData):

Canonical link: <a href="https://commits.webkit.org/261133@main">https://commits.webkit.org/261133@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/7fd42064f8e00e02edbcb2df94572aa058e3bc79

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110640 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19721 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43215 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1981 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119488 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/114599 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21133 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10838 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/1514 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116381 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/15736 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/98926 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102958 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/97697 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30593 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44071 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12367 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31931 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/85975 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12917 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/8883 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18284 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51548 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7712 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14856 "Built successfully") | | | 
| | | | |

### CODE REVIEW

File: Source/WebCore/WebCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	archiveVersion = 1;
 	classes = {
 	};
-	objectVersion = 54;
+	objectVersion = 51;
```
Comment (achristensen07) [inline Source/WebCore/WebCore.xcodeproj/project.pbxproj]: This is unrelated.  Let's not change this.

File: Source/WebCore/platform/graphics/win/FontCacheWin.cpp [hunk 2]
```diff
@@ -647,37 +638,26 @@ Vector<FontSelectionCapabilities> FontCache::getFontSelectionCapabilitiesInFamil
 
 std::unique_ptr<FontPlatformData> FontCache::createFontPlatformData(const FontDescription& fontDescription, const AtomString& family, const FontCreationContext&)
 {
-    bool isLucidaGrande = equalLettersIgnoringASCIICase(family, "lucida grande"_s);
-
-    bool useGDI = fontDescription.renderingMode() == FontRenderingMode::Alternate && !isLucidaGrande;
-
     // The logical size constant is 32. We do this for subpixel precision when rendering using Uniscribe.
     // This masks rounding errors related to the HFONT metrics being  different from the CGFont metrics.
     // FIXME: We will eventually want subpixel precision for GDI mode, but the scaled rendering doesn't
```
Comment (achristensen07) [inline Source/WebCore/platform/graphics/win/FontCacheWin.cpp]: Remove

File: Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp [hunk 3]
```diff
@@ -130,9 +127,9 @@ RefPtr<Font> Font::platformCreateScaledFont(const FontDescription& fontDescripti
 
     LOGFONT winfont;
     GetObject(m_platformData.hfont(), sizeof(LOGFONT), &winfont);
-    winfont.lfHeight = -lroundf(scaledSize * (m_platformData.useGDI() ? 1 : 32));
+    winfont.lfHeight = -lroundf(scaledSize * 32);
```
Comment (achristensen07) [inline Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp]: Could this 32 be given the same name as the 32 used earlier?  Fewer magic numbers.  Or at least a comment explaining what 32 means here.

**Review by achristensen07 — APPROVED** [review]

File: Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp (hunk 3 above)
Comment (fujii) [reply Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp]: Good point. I didn't understand it. `* 32` is for CG fonts. WinCairo might should've used the useGDI code path. I should study before removing.

File: Source/WebCore/WebCore.xcodeproj/project.pbxproj (hunk 1 above)
Comment (fujii) [reply Source/WebCore/WebCore.xcodeproj/project.pbxproj]: Fixed

File: Source/WebCore/platform/graphics/win/FontCacheWin.cpp (hunk 2 above)
Comment (fujii) [reply Source/WebCore/platform/graphics/win/FontCacheWin.cpp]: Done.

File: Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp (hunk 3 above)
Comment (fujii) [reply Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp]: Due to [cairo's bug](https://gitlab.freedesktop.org/cairo/cairo/-/issues/3), I can't remove the `* 32` at the moment. So, I added a constant `cWindowsFontScaleFactor`.
